File: File Templates/Viper View Controller.xctemplate/Objective-C/___FILEBASENAME___Presenter.m
//___FILEHEADER___
 
#import "___VARIABLE_ModuleName___Presenter.h"

@implementation ___VARIABLE_ModuleName___Presenter

- (instancetype)initWithInterface:(id<___VARIABLE_ModuleName___ViewProtocol>)interface interactor:(id<___VARIABLE_ModuleName___InteractorInputProtocol>)interactor router:(id<___VARIABLE_ModuleName___WireframeProtocol>)router
{
    if (self = [super init])
    {
        self.view = interface;
        self.interactor = interactor;
        self.router = router;
        [self.interactor setPresenter:self];
    }
    return self;
}

@end

File: File Templates/VIPER/Viper View Controller.xctemplate/Swift/___FILEBASENAME___Presenter.swift
//___FILEHEADER___

import UIKit

final class ___VARIABLE_ModuleName___Presenter: ___VARIABLE_ModuleName___PresenterProtocol  {
    
    // MARK: - Accessable

    weak var view: ___VARIABLE_ModuleName___ViewProtocol?
    var interactor: ___VARIABLE_ModuleName___InteractorInputProtocol?
    var wireFrame: ___VARIABLE_ModuleName___RouteProtocol?

    // MARK: - Private

    // MARK: - Init
    
    init() { }
}

// MARK: - Extension INTERACTOR -> PRESENTER

extension ___VARIABLE_ModuleName___Presenter: ___VARIABLE_ModuleName___InteractorOutputProtocol { }
